homeassistant/components/vodafone_station/button.py

```diff
@@ -63,6 +63,14 @@ class VodafoneStationEntityDescription(ButtonEntityDescription):
         ),
         is_suitable=lambda info: info.get("vf_internet_key_online_since") != "",
     ),
+    VodafoneStationEntityDescription(
+        key="cleanup",
+        translation_key="cleanup",
+        entity_category=EntityCategory.CONFIG,
+        press_action=lambda coordinator: coordinator.cleanup_device_tracker_entities(),
+        is_suitable=lambda _: True,
+        icon="",
+    ),
 )
```

Comment:
This seems invalid, this entity does not represent a device feature.

Reply:
In the old days of fritz it was a service, then we moved to a button there.
This is why here I coded it as a button.

Any other suggestion ?

Reply:
I have no other suggestion, besides it shouldn't be an entity.

Currently, entities and devices should represent the device or services it implements.

Reply:
IC, but then we should maybe implement an automatic cleanup at every init or similar ?

Reply:
@chemelli74 you can implement automatic cleanup at runtime?

I did manage to implement this for plugwise, at the moment under testing in plugwise-beta custom_component.
The code is here: https://github.com/plugwise/plugwise-beta
PR's of interest: https://github.com/plugwise/plugwise-beta/pull/641, https://github.com/plugwise/plugwise-beta/pull/627
Please note, for Plugwise there can be more than 1 device, so I had to make sure that the removed device(s) connected to the Plugwise gateway-of-interest were removed and not all the devices connected to another Plugwise gateway.

And, again, I found there is no need to clean up the entity_registry. I had code made for it but the lines that did the actual deletion were never hit, as shown by the detailed test-coverage report.

Reply:
> And, again, I found there is no need to clean up the entity_registry.

yes, entities related to an device entry which is going to be deleted, will also be deleted - but we don't delete any device entry, we just remove the current config entry from this device entry - if this was the last config entry, than the device registry will automatically delete this device entry and those all linked entities. This means, as long as the device entry is not deleted (_because it is still kinked to another config entry_) the integration have to ensure the removing of the corresponding entities

Reply:
OK a device can have more than one config_entry, that wasn't clear to me.

Reply:
> OK a device can have more than one config_entry, that wasn't clear to me.

see [Device Registry](https://developers.home-assistant.io/docs/device_registry_index) in the dev docs 😉 


> IC, but then we should maybe implement an automatic cleanup at every init or similar ?

Hi Simone (@chemelli74)
maybe compare on each coordinator update if the list of mac addresses differ - if yes, trigger the cleanup mechanic.
Further after coordinator setup (_when all current data are fetched_) also trigger the cleanup once.
I've done similar in the `fritzbox` integration (_#114601_)

Reply:
Button avoided, new approach based on https://github.com/home-assistant/core/pull/116024#issuecomment-2438547777